Lib/multiprocessing/synchronize.py

```diff
@@ -59,6 +60,10 @@ def __init__(self, kind, value, maxvalue, *, ctx):
                     unlink_now)
             except FileExistsError:
                 pass
+            except OSError as e:
+                if e.errno == errno.ENOSPC:
+                    raise Exception("There is not enough space in the"
```

Comment:
This should be a more specific error, perhaps `ProcessError`.

Reply:
I agree with Adam here.